### PR TITLE
[16.0][MIG] account_fiscal_position_autodetect_optional_vies: Migration to 16.0

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: account_fiscal_position_autodetect_optional_vies/README.rst [hunk 1]
```diff
@@ -98,6 +102,6 @@ Current `maintainer <https://odoo-community.org/page/maintainer-role>`__:
 
 |maintainer-victoralmau| 
 
-This module is part of the `OCA/account-fiscal-rule <https://github.com/OCA/account-fiscal-rule/tree/15.0/account_fiscal_position_autodetect_optional_vies>`_ project on GitHub.
+This module is part of the `OCA/account-fiscal-rule <https://github.com/OCA/account-fiscal-rule/tree/16.0/account_fiscal_position_autodetect_optional_vies>`_ project on GitHub.
```
Comment (ljsalvatierra-factorlibre) [inline account_fiscal_position_autodetect_optional_vies/README.rst:105]: Did you modified this manually? No need.
Same for all the other modifications `15` -> `16`.

File: account_fiscal_position_autodetect_optional_vies/models/account_fiscal_position.py [hunk 2]
```diff
@@ -14,32 +14,32 @@ class AccountFiscalPosition(models.Model):
     )
 
     @api.model
-    def search(self, args, offset=0, limit=None, order=None, count=False):
-        if "vat_vies_required" in self.env.context:
-            args = expression.AND(
+    def search(self, domain, offset=0, limit=None, order=None, count=False):
+        if self.env.context.get("vat_vies_required"):
```
Comment (ljsalvatierra-factorlibre) [inline account_fiscal_position_autodetect_optional_vies/models/account_fiscal_position.py]: Previous check was better, we only need to modify the domain if `vat_vies_required` is in the context.

File: account_fiscal_position_autodetect_optional_vies/models/account_fiscal_position.py [hunk 3]
```diff
@@ -14,32 +14,32 @@ class AccountFiscalPosition(models.Model):
...
         )
 
     @api.model
-    def get_fiscal_position(self, partner_id, delivery_id=None):
+    def _get_fiscal_position(self, partner, delivery=None):
         _self = self
-        if partner_id:
-            partner = self.env["res.partner"].browse(delivery_id or partner_id)
+        if partner:
+            partner = partner or delivery
```
Comment (ljsalvatierra-factorlibre) [inline account_fiscal_position_autodetect_optional_vies/models/account_fiscal_position.py]: It should have more priority delivery than partner.

**Review by ljsalvatierra-factorlibre — CHANGES_REQUESTED** [review]

File: account_fiscal_position_autodetect_optional_vies/README.rst (hunk 1 above)
Comment (ghost) [reply account_fiscal_position_autodetect_optional_vies/README.rst:105]: No, "pre-commit" made these changes.

File: account_fiscal_position_autodetect_optional_vies/models/account_fiscal_position.py (hunk 2 above)
Comment (ghost) [reply account_fiscal_position_autodetect_optional_vies/models/account_fiscal_position.py]: Changed!

File: account_fiscal_position_autodetect_optional_vies/models/account_fiscal_position.py (hunk 3 above)
Comment (ghost) [reply account_fiscal_position_autodetect_optional_vies/models/account_fiscal_position.py]: Changed!

**Review by ljsalvatierra-factorlibre — APPROVED** [review]

**Review by SoniaViciana — APPROVED** [review]
Functional review LGTM!

**Comment by dreispt** [comment]
/ocabot migrate account_fiscal_position_autodetect_optional_vies
/ocabot merge nobump

**Comment by ghost** [comment]
> /ocabot migrate account_fiscal_position_autodetect_optional_vies /ocabot merge nobump

Hi @dreispt,

Could you re-try merging the migration?

Thank you!

**Comment by aritzolea** [comment]
We found a bug in the migration, don't merge it please until it is solved.

**Comment by dreispt** [comment]
/ocabot migration account_fiscal_position_autodetect_optional_vies

**Comment by dreispt** [comment]
I started the merge but then stopped it.

**Comment by aritzolea** [comment]
It can now be merged @dreispt , thanks

**Comment by rafaelbn** [comment]
Doesn't look this command is working:

/ocabot migrate account_fiscal_position_autodetect_optional_vies

**Comment by rafaelbn** [comment]
It doesn't look this command is working:

/ocabot migration account_fiscal_position_autodetect_optional_vies

**Comment by Tisho99** [comment]
Hi @ACheung-FactorLibre @aritzolea @dreispt @rafaelbn 

I am interested in migrating this module to version 17. Is this PR going to be merged? What it is needed to it?

**Review by Tisho99 — APPROVED** [review]
LGTM technical review

**Pull request merged**